src/dkim/parse.rs

```diff
@@ -241,24 +240,14 @@ impl SignatureParser for Iter<'_, u8> {
     }
 }
 
-enum KeyType {
-    Rsa,
-    Ed25519,
-    None,
-}
-
 impl TxtRecordParser for DomainKey {
     #[allow(clippy::while_let_on_iterator)]
     fn parse(header: &[u8]) -> crate::Result<Self> {
         let header_len = header.len();
         let mut header = header.iter();
-        let mut record = DomainKey {
-            v: Version::V1,
-            p: PublicKey::Revoked,
-            f: 0,
-        };
-        let mut k = KeyType::None;
-        let mut public_key = Vec::new();
+        let mut flags = 0;
+        let mut key_type = VerifyingKeyType::Rsa;
```

Comment:
The key type is actually optional, so this seems better. I removed the `DomainKey` here to avoid the invalid state, also prefer to be explicit about `Option` for the `public_key` instead of letting it decide based on `is_empty()`.